### PR TITLE
New Devil Obligation: Dance Off

### PR DESCRIPTION
**Pull request by RemieRichards**
:cl:
add: Devils may now spawn with an obligation to accept dance off challanges, if they have this obligation they also gain a spell to summon/unsummon a 3x3 dance floor at will.
/:cl:

## Misc:
* New Smoke Types
* It's much easier (and nicer) to make a devil's spell irremovable

ShutupAndDance

### CODE REVIEW

**Comment by ma44** [comment]
How does it determine who wins the dance off?

**Comment by RemieRichards** [comment]
Same as with the Musical Fiddle Duel.
Up to the participants/crowd to decide.

**Comment by Xhuis** [comment]
Each day we stray further from God's light

**Comment by lzimann** [comment]
i only challenge devils to a rock off

**Comment by Iamgoofball** [comment]
Just make it DDR

**Comment by Dagdammit** [comment]
and have the actual input arrows fall from the top of the screen, but they're actual objects and any you miss hit the station like meteors

File: code/game/turfs/simulated/floor/light_floor.dm [hunk 1]
```diff
@@ -42,6 +43,8 @@
 	..()
 
 /turf/open/floor/light/attack_hand(mob/user)
+	if(!can_modifiy_colour)
```
Comment (coiax) [inline code/game/turfs/simulated/floor/light_floor.dm]: modifiy

**Comment by RemieRichards** [comment]
@Xhuis and into Satan's funky disco!

**Comment by RemieRichards** [comment]
also I will consider making it actual DDR.
in that sense, the spell will become targeted, and there will be two pads, one per person.
and it would just be DDR basically.
would need like, 1/2 songs to go with it though, otherwise it's just silent DDR and that's weird.

tl;dr get me songs and I'll try DDR.

**Comment by AdamElTablawy** [comment]
we are number one

File: code/game/turfs/simulated/floor/light_floor.dm [hunk 2]
```diff
@@ -67,3 +70,23 @@
 			user << "<span class='notice'>You replace the light bulb.</span>"
 		else
 			user << "<span class='notice'>The lightbulb seems fine, no need to replace it.</span>"
+
+
+//Cycles through all of the colours
+/turf/open/floor/light/colour_cycle
+	coloredlights = list("cycle_all")
+	can_modifiy_colour = FALSE
```
Comment (PKPenguin321) [inline code/game/turfs/simulated/floor/light_floor.dm]: missed one

**Comment by AnturK** [comment]
Make it grab the tempo from the closest playing instrument for DDR. I guess you'd need a third person then but eh.

**Comment by KingPhilipIII** [comment]
Devil better say something clever when casting the spell, like he does with the dance floor.

**Comment by jughu** [comment]
Better give the devil the ability to return the person soul or victims souls if he loses
